clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -128,6 +218,37 @@ void collectCmakeScripts(StringRef InRoot, StringRef OutRoot,
   }
 }
 
+// apply patter rewrite rules to migrate cmake script file
+static void
+applyPatternRewriterToCmakeScriptFile(const std::string &InputString,
+                                      llvm::raw_os_ostream &Stream) {
+  std::string LineEndingString;
+  // pattern_rewriter require the input file to be LF
+  bool IsCRLF = fixLineEndings(InputString, LineEndingString);
+
+  // Convert cmake command to lower case in cmake script files
+  LineEndingString = convertCmakeCommandsToLower(LineEndingString);
+
+  std::map<std::string, std::string> VariablesMap;
+  parseVariable(LineEndingString, VariablesMap);
+  cmakeSyntaxProcessed(LineEndingString, VariablesMap);
```

Comment:
Please add comment on: Line 234 and L236.   what happen if both cmakeSyntaxProcessed and applyPatternRewriter process on same cmake syntax.